### PR TITLE
cleanup build scripts.

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: buildscripts/checkdeps.sh [hunk 1]
```diff
@@ -23,12 +23,19 @@ _init() {
...
+    ARCH=$(uname -m)
 }
 
+## FIXME:
+## In OSX, 'readlink -f' option does not exist, hence
+## we have our own readlink -f behaviour here.
+## Once OSX has the option, below function is good enough.
+## 
+## readlink() {
+##     return /bin/readlink -f "$1"
```
Comment (harshavardhana) [inline buildscripts/checkdeps.sh:36]: why is this removed?

File: buildscripts/checkdeps.sh [hunk 2]
```diff
@@ -50,152 +57,114 @@ readlink() {
...
-            ver2[i]=0
-        fi
-        if ((10#${ver1[i]} > 10#${ver2[i]})); then
-
+## FIXME:
+## In OSX, 'sort -V' option does not exist, hence
+## we have our own version compare function.
+## Once OSX has the option, below function is good enough.
+##
+## check_minimum_version() {
```
Comment (harshavardhana) [inline buildscripts/checkdeps.sh:65]: Can you validate this on OS X?

File: buildscripts/checkdeps.sh [hunk 3]
```diff
@@ -50,152 +57,114 @@ readlink() {
...
-        exit -1
-    fi
+assert_is_supported_arch() {
+    case "${ARCH}" in
+        x86_64 | amd64 | aarch64 | arm* )
+	    return
+            ;;
+        *)
+	    echo "ERROR"
+            echo "Arch '${ARCH}' not supported."
```
Comment (harshavardhana) [inline buildscripts/checkdeps.sh:92]: additional spaces here..

File: buildscripts/checkdeps.sh [hunk 4]
```diff
@@ -50,152 +57,114 @@ readlink() {
...
+##     versions=($(echo -e "$1\n$2" | sort -V))
+##     return [ "$1" == "${versions[0]}" ]
+## }
+##
+check_minimum_version() {
+    IFS='.' read -r -a varray1 <<< "$1"
+    IFS='.' read -r -a varray2 <<< "$2"
+
+    for i in "${!varray1[@]}"; do
+        if [[ ${varray1[i]} < ${varray2[i]} ]]; then
```
Comment (harshavardhana) [inline buildscripts/checkdeps.sh:75]: why only assuming that we are not interested in the minor version?

File: buildscripts/checkdeps.sh [hunk 5]
```diff
@@ -50,152 +57,114 @@ readlink() {
...
+		exit 1
+	    fi
+	    return
             ;;
-        "*")
-            echo "Exiting.. unsupported operating system found"
-            exit 1;
+	*)
+	    echo "ERROR"
+	    echo "OS '${KNAME}' not supported."
```
Comment (harshavardhana) [inline buildscripts/checkdeps.sh]:
```
echo "OS '${KNAME}'  is not supported."
```

File: buildscripts/checkdeps.sh (hunk 1 above)
Comment (balamurugana) [reply buildscripts/checkdeps.sh:36]: Its a comment.

File: buildscripts/checkdeps.sh (hunk 2 above)
Comment (balamurugana) [reply buildscripts/checkdeps.sh:65]: Done

File: buildscripts/checkdeps.sh (hunk 4 above)
Comment (balamurugana) [reply buildscripts/checkdeps.sh:75]: As we loop through bash array, its checked.

File: buildscripts/checkdeps.sh (hunk 3 above)
Comment (balamurugana) [reply buildscripts/checkdeps.sh:92]: Done

File: buildscripts/checkdeps.sh (hunk 5 above)
Comment (balamurugana) [reply buildscripts/checkdeps.sh]: Done
